src/basic/os-util.c

```diff
@@ -317,3 +317,35 @@ int load_extension_release_pairs(const char *root, const char *extension, char *
 
         return load_env_file_pairs(f, p, ret);
 }
+
+int os_release_support_ended(const char *support_end, bool quiet) {
+        _cleanup_free_ char *t = NULL;
+        int r;
+
+        if (!support_end) {
+                r = parse_os_release(NULL,
+                                     "SUPPORT_END", &t);
+                if (r < 0)
+                        return log_full_errno(r == -ENOENT || quiet ? LOG_DEBUG : LOG_WARNING, r,
+                                              "Failed to read os-release file, ignoring: %m");
+                if (!t)
+                        return false;  /* no end date defined */
+
+                support_end = t;
```

Comment:
Did you forget `TAKE_PTR`?

Reply:
No, it's supposed to be cleaned up. Either we get a param from the outside, or we read it in.

Reply:
I renamed it and added a comment.

Reply:
> I renamed it and added a comment.

That wasn't really necessary. For some reason I looked at support_end and thought it was a `char **`.